tests/integration/deploy_txn_tests.rs

```diff
@@ -358,3 +358,144 @@ async fn deploy_via_invoke_no_calldata_cairo1_account(
     .await
     .expect("OS run failed");
 }
+
+#[rstest]
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn deploy_cairo0_without_proxy(block_context: BlockContext, max_fee: Fee) {
+    let test_proxy_account = load_cairo0_feature_contract("test_proxy_account");
+    let class_hash = class_hash!("0x9ea83ef3294e2b3e2655642e8340dfacc330175a70fe79367e199215b7ffa1");
+
+    let args: Vec<Felt252> = vec![];
+
+    let ctor_calldata = Calldata(args.into());
+    let deployed_contract_address = calculate_contract_address(
+        ContractAddressSalt::default(),
+        class_hash,
+        &ctor_calldata,
+        contract_address!("0x0"),
+    )
+    .expect("Failed to calculate the contract address");
+
+    let initial_state = StarknetStateBuilder::new(&block_context)
+        .deploy_cairo0_contract(test_proxy_account.0, test_proxy_account.1)
+        .fund_account(deployed_contract_address, BALANCE, BALANCE)
+        .set_default_balance(BALANCE, BALANCE)
+        .build()
+        .await;
+
+    let tx_version = TransactionVersion::ONE;
+    let mut nonce_manager = NonceManager::default();
+
+    let test_proxy_account = initial_state.deployed_cairo0_contracts.get("test_proxy_account").unwrap();
+
+    let deployed_account_class_hash = test_proxy_account.declaration.class_hash;
+    // Sanity check, as we hardcode the class hash in the fixture we verify that we have
+    // the right one.
+    assert_eq!(class_hash, deployed_account_class_hash);
+
+    let deploy_account_tx = AccountTransaction::DeployAccount(deploy_account_tx(
+        deploy_account_tx_args! {
+            class_hash: deployed_account_class_hash,
+            max_fee,
+            contract_address_salt: ContractAddressSalt::default(),
+            version: tx_version,
+            constructor_calldata: ctor_calldata,
+            signature:  TransactionSignature([
+                "0x0",
+                "0x0",
+            ].into_iter()
+            .map(Felt252::from_hex_unchecked)
+            .collect()),
+        },
+        &mut nonce_manager,
+    ));
+
+    let txs = vec![deploy_account_tx].into_iter().map(Into::into).collect();
+    let _result = execute_txs_and_run_os(
+        crate::common::DEFAULT_COMPILED_OS,
+        initial_state.cached_state,
+        block_context,
+        txs,
+        initial_state.cairo0_compiled_classes,
+        initial_state.cairo1_compiled_classes,
+        HashMap::default(),
+    )
+    .await
+    .expect("OS run failed");
+}
+
+#[rstest]
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn deploy_cairo0_via_proxy(block_context: BlockContext, max_fee: Fee) {
```

Comment:
Does the proxy impact this bug at all? I might apply my same concerns above if not, this is likely to add confusion in the future.

Reply:
Resolved by https://github.com/keep-starknet-strange/snos/pull/431/commits/c4726a7c44e2a64d7156aeaea7075d52c14ee601